TWC_Backend/TWC_Services/DBService/Services/DBProjectService.cs

```diff
@@ -0,0 +1,227 @@
+﻿using Microsoft.EntityFrameworkCore;
+using TWC_DatabaseLayer;
+using TWC_DatabaseLayer.DTOs;
+using TWC_DatabaseLayer.Models;
+using TWC_Services.DBService.Interfaces;
+
+namespace TWC_Services.DBService.Services
+{
+    public class DBProjectService : IDBProjectService
+    {
+        private DataContext _context;
+
+        public DBProjectService(DataContext context) {
+            _context = context; 
+        }
+        public async Task<Project> CreateProjectAsync(ProjectCreationDTO project)
+        {
+            try
+            {
+                var newProject = new Project()
+                {
+                    Name = project.Name,
+                    MaxMembers = project.MaxMembers,
+                    Description = project.Description,
+                    Members = new List<ProjectMember>(),
+                    Tags = new List<ProjectTag>(),
+                };
+
+                await _context.Projects.AddAsync(newProject);
+                await _context.SaveChangesAsync();
+
+                User owner = await _context.Users.FindAsync(project.CreatorId);
+
+                if (owner == null)
+                {
+                    return null;
+                }
+
+                newProject.Members.Add(new ProjectMember()
+                {
+                    IsOwner = true,
+                    ProjectId = newProject.Id,
+                    UserId = owner.Id,
+                });
+
+                await _context.SaveChangesAsync();
+
+                foreach (string tagDTO in project.Tags)
+                {
+                    Tag tag = await _context.Tags.SingleOrDefaultAsync(x => x.Name == tagDTO);
+
+                    if(tag == null)
+                    {
+                        return null;
+                    }
```

Comment:
At first I thought you were using List<string> because you wanted to add new tags. But if you only want to add existing ones, then isn't it easier to just pass the List<Tag>? At the front end, when selecting, we will record from existing tags, and they will already have the id. You can only check if the tag exists, if it's nessesary

Reply:
I need tag id to add it to the project